### PR TITLE
Minimize duplicate files

### PR DESCRIPTION
**Pull request by akaviaLab**
* [X] description of feature/fix
Some files were present in src/cobra/data as example files as well as tests/data.
I removed duplicate files, set up the tests to use either the src/cobra/data file or tests/data file as appropriate.
I also updated update_pickles.py to create the src/cobra/data files when run.

In addition, I also updated SBML, JSON, MAT to make sure they can use Path as input.
[X] Tests modified
test_json: test_load_json_model, test_json::test_save_json_model 
Modified to use str, file handle and Path, make sure all methods work
test_mat::test_load_matlab_model, test_mat::test_save_matlab_model
Modified to use str, file handle and Path, make sure all methods work. Parametized with fixture request.

In general, tests and code were modified to minimize use of os, replacing it with pathlib.Path

* [ ] add an entry to the [next release](../release-notes/next-release.md)

### CODE REVIEW

File: src/cobra/io/sbml.py [hunk 1]
```diff
@@ -488,35 +488,27 @@ def _get_doc_from_filename(filename: Union[str, IO]) -> "libsbml.SBMLDocument":
...
-            )  # noqa: E501 type: libsbml.SBMLDocument
+    if isinstance(filename, Path) and {".bz2", ".gz"}.isdisjoint(filename.suffixes):
+        doc: libsbml.SBMLDocument = libsbml.readSBMLFromString(filename.read_text())
+    elif isinstance(filename, Path) and {".bz2", ".gz"}.intersection(filename.suffixes):
+        doc: libsbml.SBMLDocument = libsbml.readSBMLFromFile(str(filename))
+    elif isinstance(filename, str):
+        if "<sbml" in filename:
+            doc: libsbml.SBMLDocument = libsbml.readSBMLFromString(filename)
+        elif (
+            ("win" in platform) and (len(filename) < 260) or "win" not in platform
```
Comment (cdiener) [inline src/cobra/io/sbml.py:499]: This seems a very unlikely edge case. Are there many SBML models whose filename is longer than 260 characters?

**Review by cdiener — COMMENTED** [review]
Looks good. One small question about the Windows edge case.

**Comment by akaviaLab** [comment]
I don't know, but the original sbml.py had it. I don't want to remove it
without a good reason.

**Review by cdiener — APPROVED** [review]
Makes sense.

**Comment by akaviaLab** [comment]
ping @matthiaskoenig 
Can you approve this as well, please? It touches on SBML

**Comment by akaviaLab** [comment]
@cdiener - can we merge please?

**Pull request merged**